src/contract.ts

```diff
@@ -25,24 +50,64 @@ function parseFelt(candidate: string): BN {
   }
 }
 
+function buildCall(contract: Contract, functionAbi: FunctionAbi): AsyncContractFunction {
+  return async function (args: Args, blockIdentifier: BlockIdentifier): Promise<any> {
+    return contract.call(functionAbi.name, args, blockIdentifier);
+  };
+}
+
+function buildInvoke(contract: Contract, functionAbi: FunctionAbi): AsyncContractFunction {
+  return async function (args: Args): Promise<any> {
+    return contract.invoke(functionAbi.name, args);
+  };
+}
+
+function buildDefault(contract: Contract, functionAbi: FunctionAbi): AsyncContractFunction {
+  if (functionAbi.stateMutability === 'view') {
+    return buildCall(contract, functionAbi);
+  }
+  return buildInvoke(contract, functionAbi);
+}
+
+function buildPopulate(contract: Contract, functionAbi: FunctionAbi): ContractFunction {
+  return function (args: Args): any {
+    return populateTransaction(contract, functionAbi, args);
+  };
+}
+
+function buildEstimate(contract: Contract, functionAbi: FunctionAbi): ContractFunction {
+  return function (args: Args): any {
+    return contract.estimate(functionAbi.name, args);
+  };
+}
+
 export class Contract {
   connectedTo: string | null = null;
```

Comment:
as suggested by @dhruvkelawala 
```suggestion
  address: string | null = null;
```

maybe we should also add readonly and public to all properties?